include/kernels/dslash_clover_helper.cuh

```diff
@@ -214,12 +220,12 @@ namespace quda {
         BACKWARDS = 1
       };
 
-      auto swizzle = [&](half_fermion x[2], int chirality, swizzle_direction dir) {
-        if (chirality == 0) cache.save_y(x[1], dir);
-        else                cache.save_y(x[0], 1 - dir);
+      auto swizzle = [&](half_fermion x[2], int chirality, swizzle_direction) {
+        if (chirality == 0) cache.save_y(x[1], target::thread_idx().y);
+        else                cache.save_y(x[0], target::thread_idx().y);
         cache.sync();
-        if (chirality == 0) x[1] = cache.load_y(1 - dir);
-        else                x[0] = cache.load_y(dir);
+        if (chirality == 0) x[1] = cache.load_y(target::thread_idx().y + 1);
+        else                x[0] = cache.load_y(target::thread_idx().y - 1);
       };
 
       swizzle(in_chi, chirality, FORWARDS); // apply the flavor-chirality swizzle between threads
```

Comment:
Random question: `swizzle_direction` is now an unnamed argument and is not used. Why not drop the direction from the arg list ? Forward/backwards seem to depend now only on chirality and the thread_idx.y

Reply:
[768af11](https://github.com/lattice/quda/pull/1465/commits/768af1141de96e4bfe2c2d09ecfc8926907a775e)